src/gl/canvas2dRe.re

```diff
@@ -88,20 +101,49 @@ external lineJoin : t => LineJoin.t => unit = "" [@@bs.set];
 external miterLimit : t => float => unit = "" [@@bs.set];
 
 /* Colors, Styles, and Shadows */
-external strokeStyle : t => string => unit = "" [@@bs.set];
-external fillStyle : t => string => unit = "" [@@bs.set];
+external setFillStyle : t => 'a => unit = "" [@@bs.set];
+external setStrokeStyle: t => 'a => unit = "" [@@bs.set];
+
+/* in re unused warnings
+  awaiting release of https://github.com/bloomberg/bucklescript/issues/1656
+  to just use [@@bs.set] directly with an ignored (style a) */
+let setStrokeStyle (type a) (ctx: t) (s: style a) (v: a) =>
+  setStrokeStyle ctx v;
+
+let setFillStyle (type a) (ctx: t) (s: style a) (v: a) =>
+  setFillStyle ctx v;
+
+external canvasGradient : 'a = "CanvasGradient" [@@bs.val];
+let instanceOf a cxr => [%bs.raw{|function(x,y) {return x instanceof y}|}] a cxr;
+
+let reifyStyle (type a) (x: 'a): (style a, a) =>
+  (if (Js.typeof x == "string") {
+    Obj.magic String
+  } else if (instanceOf x canvasGradient) {
+    Obj.magic Gradient
+  } else {
+    Obj.magic Pattern
```

Comment:
This assumes that these will be the only possible alternatives for the lifetime of this code. But since web standards tend to change I think it would be a good idea to run a check on `Pattern` too, and explicitly raise an `Invalid_argument`  exception if the unexpected happens. Just to be safe :)